src/deserialize_sequence.cc

```diff
@@ -94,48 +94,48 @@ void extract_type(gul14::string_view extract, Step& step)
         case "end"_sh:
             step.set_type(Step::type_end); break;
         default:
-            throw Error(gul14::cat("type: unable to parse (\"", escape(keyword), "\")"));
+            throw Error(gul17::cat("type: unable to parse (\"", escape(keyword), "\")"));
     }
 }
 
-void extract_label(gul14::string_view extract, Step& step)
+void extract_label(std::string_view extract, Step& step)
 {
-    step.set_label(gul14::unescape(gul14::trim_sv(extract)));
+    step.set_label(gul17::unescape(gul17::trim_sv(extract)));
 }
 
-void extract_context_variable_names(gul14::string_view extract, Step& step)
+void extract_context_variable_names(std::string_view extract, Step& step)
 {
-    extract = gul14::trim_sv(extract);
-    if (not gul14::starts_with(extract, "["))
+    extract = gul17::trim_sv(extract);
+    if (not gul17::starts_with(extract, "["))
         throw Error("context variable names: cannot find leading '['");
     extract.remove_prefix(1);
 
     auto end = extract.find("]");
-    if (end == gul14::string_view::npos)
+    if (end == std::string_view::npos)
         throw Error("context variable names: cannot find trailing ']'");
     VariableNames variableNames{};
-    for (auto variable: gul14::split(extract.substr(0, end), std::regex{"[ \t]*,[ \t]*"})) {
-        variable = gul14::trim(variable);
+    for (auto variable: gul17::split(extract.substr(0, end), std::regex{"[ \t]*,[ \t]*"})) {
+        variable = gul17::trim(variable);
         if (not variable.empty())
             variableNames.emplace(variable);
     }
     if (not variableNames.empty())
         step.set_used_context_variable_names(variableNames);
 }
 
-TimePoint extract_time(const std::string& issue, gul14::string_view extract)
+TimePoint extract_time(const std::string& issue, std::string_view extract)
 {
     std::tm t;
 
     if (strptime(std::string{ extract }.c_str(), "%Y-%m-%d %H:%M:%S",&t) == nullptr)
```

Comment:
This has nothing to do with this PR, but why do we construct a new string here?
Is there any possibility that
```suggestion
    if (strptime(extract.data(), "%Y-%m-%d %H:%M:%S",&t) == nullptr)
```
does not do the right thing? I can not imagine any case.

---

Ugg, the code is my code :-}

![image](https://github.com/user-attachments/assets/9f3740f1-e77b-4e50-bd5c-afe0fe0f75c9)

---

Reasoning:
- `extract` is a view to a guaranteed zero-terminated string
- `strptime()` breaks off conversion at the first char that does not match the format, or
- when the format is completely matched, ignoring all following chars

The older code used a `string const&` as `extract` parameter and `extract.substr()` (which also creates a new string) - albeit it just wanted to advance two chars :thinking: 

![image](https://github.com/user-attachments/assets/f63366a1-bc23-4692-bfd3-f6f3457bac0c)

:grimacing:

Reply:
I think this should be discussed in another issue instead of polluting this pull request. It has nothing to do with original request abut the Catch2v3 etc. migration issue.

About null-termination: I agree because the member function [std::string_view<>::data()](https://www.cppreference.com/w/cpp/string/basic_string_view/data.html) cannot guarantee null-termination. Therefore we need a temporary use of a `std::string` and return via the `c_str()` the character pointer. And here it is guaranteed that it is a null-terminated. If have another option then feel free to point to the correct path 😄 . 

I also asked ChatGPT about another solution and I think it has some thinkable code snippet with an `std::optional`:

```c++
#include <string>
#include <string_view>
#include <ctime>
#include <optional>

std::optional<std::tm> parse_time(std::string_view time_str, std::string_view format) {
    std::string null_terminated_time(time_str);  // make a safe copy
    std::tm t = {};

    char* res = strptime(null_terminated_time.c_str(), format.data(), &t);
    if (res == nullptr) {
        return std::nullopt;  // parsing failed
    }

    return t;  // success
}
```

However, the `format` can also be a source of errors :sunglasses:.

Reply:
> I also asked ChatGPT

You got me with that joke :rofl: